java/test/jmri/implementation/AbstractSensorTestBase.java

```diff
@@ -217,6 +217,27 @@ public void testSensor() throws JmriException {
         jmri.util.JUnitUtil.waitFor(()->{return t.getCommandedState() == Sensor.OFF;}, "commanded state = OFF");
         Assert.assertTrue("Sensor is ON", t.getCommandedState() == Sensor.OFF);
     }
+    
+    @Test
+    public void testSensorSetKnownState() throws JmriException {
+
+        t.setKnownState(Sensor.ACTIVE);
+        Assert.assertEquals("ACTIVE", t.describeState(Sensor.ACTIVE), t.describeState(t.getState()));
```

Comment:
It would be better to use:
`Assert.assertEquals("ACTIVE", Sensor.ACTIVE, t.getState());`

Reply:
Thanks for fixing this.

Reply:
No probs.
I find the String value a lot easier to work with so have left one commented out for when needed ;-)